File: web/src/types/group.ts
// MindCanvas P1 分组类型

export interface Group {
  id: string
  room_id: string
  name: string
  color: string
  members: string[]      // student guest_uuid 数组
  leader_uuid: string    // 空字符串 = 无组长
  sort_order: number
  zone_element_id?: string
  created_at: string
  updated_at: string
}

export interface AutoGroupRequest {
  mode: 'by_groups' | 'by_count'
  n: number
}

export interface AutoGroupResponse {
  message: string
  groups: Pick<Group, 'id' | 'name' | 'color' | 'members' | 'sort_order'>[]
}

File: server/models/group.go
package models

import "time"

// Group 课堂分组
type Group struct {
	ID            string    `json:"id"`
	RoomID        string    `json:"room_id"`
	Name          string    `json:"name"`
	Color         string    `json:"color"`
	Members       []string  `json:"members"`
	LeaderUUID    string    `json:"leader_uuid"`
	SortOrder     int       `json:"sort_order"`
	ZoneElementID *string   `json:"zone_element_id,omitempty"`
	CreatedAt     time.Time `json:"created_at"`
	UpdatedAt     time.Time `json:"updated_at"`
}

// AutoGroupRequest 自动分组请求体
// mode="by_groups": n 为组数
// mode="by_count":  n 为每组人数
type AutoGroupRequest struct {
	Mode string `json:"mode" binding:"required"`
	N    int    `json:"n"    binding:"required,min=1"`
}
